californium-core/src/main/java/org/eclipse/californium/core/network/BaseMatcher.java

```diff
@@ -143,25 +155,31 @@ public void clear() {
 	 */
 	protected final void registerObserve(final Request request) {
 
-		// We ignore blockwise request, except when this is an early negotiation
-		// (num and M is set to 0)
-		if (!request.getOptions().hasBlock2() || request.getOptions().getBlock2().getNum() == 0
-				&& !request.getOptions().getBlock2().isM()) {
+		// We ignore blockwise request, except when this is an early negotiation (num set to 0)
+		if (!request.getOptions().hasBlock2()
+				|| request.getOptions().getBlock2().getNum() == 0) {
```

Comment:
If you analyse, what happens, if M is 1, you will see, that this change is **required**.
(I added a uni-test for that, this will be included in the next commit)

```
CON [MID=44407, T=68adebdcec69a7db], GET, /test, 2:0/1/64    ----->
<-----   ACK [MID=44407, T=68adebdcec69a7db], 2.05, 2:0/1/64, size2(152)
CON [MID=44408, T=68adebdcec69a7db], GET, /test, 2:2/0/32    ----->
<-----   ACK [MID=44408, T=68adebdcec69a7db], 2.05, 2:2/1/32
CON [MID=44409, T=68adebdcec69a7db], GET, /test, 2:3/0/32    ----->
<-----   ACK [MID=44409, T=68adebdcec69a7db], 2.05, 2:3/1/32
CON [MID=44410, T=68adebdcec69a7db], GET, /test, 2:4/0/32    ----->
<-----   ACK [MID=44410, T=68adebdcec69a7db], 2.05, 2:4/0/32
```

Otherwise, if we want to change the behaviour and ensure, that request with BLOCK1, M 1 is reported as error, this will require more changes and I would prefer to have this in a different issue/PR.

So, this fixes this current implementation.

Reply:
it make sense